src/components/Input/Input.tsx

```diff
@@ -84,27 +83,37 @@ export const Input = React.forwardRef<HTMLInputElement, Props>(
         const isLastPassDisabled = type !== 'password';
 
         return (
-            <>
+            <div
+                {...block({
+                    withIcon: leadingIcon,
+                    context,
+                    size,
+                    isBlock,
+                    disabled,
+                    readOnly,
+                })}
```

Comment:
either the whole `rest` or at least `classname` from there should be passed here, so that classes defined on the application side are also applied. 
Also probably `style` should be applied to this element, rather then to input (even though other rest attributes do belong to there)

Reply:
Sorry, I wasn't clear. The `...rest` should go into the list of thing `block` called with. It knows how to combine the classname there. It is indeed a good idea to pass the `styles` here though. I don't think block manages that.